cpp/include/cuvs/neighbors/graph_build_types.hpp

```diff
@@ -56,10 +56,32 @@ struct ivf_pq_params {
   ivf_pq_params(raft::matrix_extent<int64_t> dataset_extents,
                 cuvs::distance::DistanceType metric = cuvs::distance::DistanceType::L2Expanded)
   {
-    build_params = cuvs::neighbors::ivf_pq::index_params::from_dataset(dataset_extents, metric);
+    build_params    = ivf_pq::index_params::from_dataset(dataset_extents, metric);
```

Comment:
I really like the `from_dataset` factory method. Could we just expose that here too? It's more explicit and follows our consistent paradigm of using factory functions/methods for object construction instead of relying on constructors (and it helps protect us against changes to the class hierarchy, should we make them in the future).